File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    a = [0]*10
    b = [0]*10
    for n in X:
        a[int(n)] += 1
    for n in Y:
        b[int(n)] += 1
    
    
    result = ""
    for i in range(9, 0, -1):
        m = min(a[i], b[i])
        if m != 0:
            result += str(i)*m
    
    m = min(a[0], b[0])
    if result == "":
        if m > 0:
            return "0"
        return "-1"
    return result + "0"*m

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
import java.util.*;

class Solution {
    public String solution(String X, String Y) {
        int[] freqX = new int[10];
        int[] freqY = new int[10];

        for (char c : X.toCharArray()) {
            freqX[c - '0']++;
        }
        for (char c : Y.toCharArray()) {
            freqY[c - '0']++;
        }

        StringBuilder sb = new StringBuilder();

        for (int i = 9; i >= 0; i--) {
            int commonCount = Math.min(freqX[i], freqY[i]);
            for (int j = 0; j < commonCount; j++) {
                sb.append(i);
            }
        }

        if (sb.length() == 0) return "-1";        
        if (sb.charAt(0) == '0') return "0";      

        return sb.toString();
    }
}
